### PR TITLE
Clean up formatting of incident response checklist

### PR DESCRIPTION
**Pull request by zachmargolis**
- Remove `- [ ]` checkboxes, they don't render well, aren't interactive
- Clean up link formatting
- Expand acronyms
- Move to "Development" section of handbook (from Team)

### CODE REVIEW

**Comment by zachmargolis** [comment]
Here's a preview in Federalist of the changes:
https://federalist-7d0b2b76-42fc-4df1-9825-8c3cd77e0a15.app.cloud.gov/preview/18f/identity-handbook/margolis-clean-up-checklist-formatting/

Compared to what we have now:
https://handbook.login.gov/articles/incident-response-checklist.html

File: _articles/incident-response-checklist.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 title: "Incident Response Checklist"
 description: "Quick reference checklist for incident response"
 layout: article
-category: "Team"
+category: "Development"
```
Comment (pauldoomgov) [inline _articles/incident-response-checklist.md:5]: Not sure WHERE to put this thing, but this is probably a better spot than "Team" or buried below.  Let's do it!

File: _articles/incident-response-checklist.md [hunk 2]
```diff
@@ -11,50 +11,50 @@ This is a quick checklist for any incident (security, privacy, outage, degraded
 
 ## Initiate
 
-* [ ] Situation Lead (SL) role assigned (or self-assigned) - Usually the team member who discovers the problem OR takes up response to outside report
-* [ ] SL: Incident declared in [#login-situation](https://gsa-tts.slack.com/archives/C5QUGUANN)
-* [ ] SL: Slack or OpsGenie used to alert additional login.gov responders - At a minimum, on call members of infrastructure, application, and security teams should be alerted
-* [ ] SL: Issue created as official record for incident: [Incident Template](https://github.com/18F/identity-security-private/issues/new?template%3Dincident_commander_template.md)  **Sensitive information like PII, secrets, etc MAY NOT be logged in the incident**
-* [ ] SL: Hangout started and shared in #login-situation
+* Situation Lead (SL) role assigned (or self-assigned): Usually the team member who discovers the problem OR takes up response to outside report
+* Situation Lead: Incident declared in [#login-situation](https://gsa-tts.slack.com/archives/C5QUGUANN)
```
Comment (pauldoomgov) [inline _articles/incident-response-checklist.md:15]: I wanted the look of checkboxes, but they didn't do anything except look dumb.  Thanks for the cleanup!

**Review by pauldoomgov — APPROVED** [review]
Thank you for the cleanup!  Much improved!

**Pull request merged**